src/pow.cpp

```diff
@@ -10,14 +10,139 @@
 #include <primitives/block.h>
 #include <uint256.h>
 #include "tinyformat.h"
+#include <boost/thread.hpp>
+
+// ProgPow
+#include <crypto/ethash/lib/ethash/endianness.hpp>
+#include <crypto/ethash/include/ethash/progpow.hpp>
+#include "crypto/ethash/helpers.hpp"
+#include "crypto/ethash/progpow_test_vectors.hpp"
+#include "validation.h"
+
+// RandomX
+#include "crypto/randomx/randomx.h"
+#include "chainparams.h"
+#include "miner.h"
+
+// Used by both CPU miner and validator
+static randomx_flags flags;
+
+// Used by CPU miner for randomx
+static uint256 mining_key_block;
+static randomx_cache *myMiningCache;
+static randomx_dataset *myMiningDataset;
+std::vector<randomx_vm*> vecRandomXVM;
+std::vector<std::thread> vecRandomXThreads;
+bool fKeyBlockedChanged = false;
+
+// Used by Validator
+static uint256 validation_key_block;
+static randomx_cache *myCacheValidating;
+static randomx_vm *myMachineValidating;
+static bool fLightCacheInited = false;
+
+bool IsRandomXLightInit()
```

Comment:
Perhaps in the future this whole bit of code should be an object? Usually, you use `if...` methods on objects, or `get`ers etc.

Perhaps leave a `TODO`

Reply:
We can. I first wanted to get it working and tested. Then we can start to optimize it

Reply:
Todo added